actors/evm/src/interpreter/system.rs

```diff
@@ -42,6 +42,8 @@ pub struct System<'r, RT: Runtime> {
     saved_state_root: Option<Cid>,
     /// Read Only context (staticcall)
     pub readonly: bool,
+    /// Restricted context; implies readonly
+    pub restricted: bool,
```

Comment:
Let's have an enum with `is_readonly`, etc.

Reply:
an enum is better, yeah.